### ISSUE TITLE
Apply PUDL entity matching framework to FERC-EIA

### ISSUE THREAD

**Issue by zschira** [catalyst-cooperative/ccai-entity-matching#108]
The final output of the CCAI project should be the complete replacement of the FERC-EIA matching in PUDL. Once #106 is complete we should be prepared to drop the framework developed here directly into PUDL (we could also add a dependency to this repo, but I believe it will be much more maintainable in PUDL). 

Out of scope:
- This does *NOT* include bringing the Splink model into PUDL, but does lay the groundwork for doing so easily in the future.
- Experiment tracking will be handled in separate PR.
- Blocking with `faiss` and comparison to existing blocking column results will be handled in separate PR (Try running modeling on blocks outputted from `faiss` clustering step and see if there's a score improvement)

```[tasklist]
## Success criteria
- [ ] FERC-EIA match serves as a template for using PUDL entity matching framework
```

```[tasklist]
## TODO
- [x] Replace `InputManager` and `Features` from `pudl.analysis.ferc1_eia_record_linkage` with `pudl.analysis.record_linkage.embed_dataframe` and name cleaner. Remove `recordlinkage` package dependency.
- [ ] Rework asset structure of FERC input so that a bunch of merges aren't performed?
- [x] Make into `graph_asset` broken down with `ops`. Should training steps be pickled and grid search cross validation be removed?
- [x] Use revert nulls functions to revert dataframe after model runs.
```
